### ISSUE TITLE
Allow CI on all branches

### ISSUE THREAD

**Issue by szabgab** [perl-actions/ci-perl-tester-helpers#14]
Now if I fork and create a branch, even if I enable CI on my forked repository the CI jobs won't run on it as they are restricted to the 'master' branch.  Does this limitation have some advantage?

**Comment by szabgab** [comment]
I have not noticed that on my repos.

**Comment by oalders** [comment]
> I have not noticed that on my repos.

You haven't noticed the duplicate CI runs on a pull request?

**Comment by szabgab** [comment]
Yeah. Only a single CI run
See here: https://github.com/szabgab/github-actions-bash/pull/2

Unless you mean one CI run for the PR and then another CI run after the PR was merged.

**Comment by oalders** [comment]
I *think* with this one you get only one CI run because it's only a PR. The push to the branch is happening on another repository. If you create a PR of your own, you might see the two runs as that would be a PR and a push to a branch on the same repo.

**Comment by szabgab** [comment]
You are right. If the PR comes from the same repository then both the `push` and the `pull-request` will trigger the CI and if later you update the branch with new commits then again the CI will run twice.

**Comment by szabgab** [comment]
Do I understand correctly that your process is either running the workflows manually via the UI or creating a PR immediately after creating a branch so you will benefit from the trigger while you work in the branch, right?

As I have not worked this way earlier I tried to research it a bit. I came up with another solution that might or might not be better: https://github.com/szabgab/github-actions-avoid-duplicate-trigger

**Comment by oalders** [comment]
> Do I understand correctly that your process is either running the workflows manually via the UI or creating a PR immediately after creating a branch so you will benefit from the trigger while you work in the branch, right?

Correct. I try to make a habit of sending pull requests even to myself. If someone wants to send a PR and mark it as a draft, the CI will still run.

> Only then, after potentially rebasing the branch, would they send a PR. This means the duplicate trigger only happens once.

This is a best case scenario, since the CI will run twice for every round of changes requested. In the grand scheme, that part is probably not confusing, but there is also an environmental cost to running CI, so I think there's a good argument for not doing the same work twice if it can easily be avoided.

Using a prefix like `internal/` is not a bad solution, but it puts the onus on the repo owners to adapt their workflow. The benefit is that it does become a little easier for newcomers. It's probably just as easy to document the CI's behaviour in a `CONTRIBUTING.md` so that new contributors can understand this part of it.

Having said that, there are cases where you don't want CI to run in a fork, like in a workflow that requires secrets which the fork won't have. So you'd need to add some additional logic to work around that as well.
